benchmarks/canny_edge_detector/functional_c#/Program.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Drawing;
+using System.Linq;
+using System.Collections.Immutable;
+
+namespace functional_c_
+{
+    class Program
+    {
+        static readonly double[] horr = {-1.0, 0.0, 1.0, -2.0, 0.0, 2.0, -1.0, 0.0, 1.0};
+        static readonly double[] verr = {1.0, 2.0, 1.0, 0.0, 0.0, 0.0, -1.0, -2.0, -1.0};
+        static readonly ImmutableList<(int x, int y, double v)> kernelHori = horr.Select((v, i) => (i % 3 - 1, i / 3 - 1, v)).ToImmutableList();
+        static readonly ImmutableList<(int x, int y, double v)> kernelVert = verr.Select((v, i) => (i % 3 - 1, i / 3 - 1, v)).ToImmutableList();
+        static readonly int weak = 100;
+        
+        static void Main(string[] args)
+        {
+            var pic = new Bitmap("benchmarks/canny_edge_detector/download.jpg");
+            var res = cannyEdge(pic);
+            System.Console.WriteLine(res.Count(p => p.w > 0));
+        }
+
+        private static ImmutableList<(int x, int y, int w)> cannyEdge(Bitmap pic)
+        {
+            var greyscaled = greyScale(pic);
+
+            var gaussianPic = BlurGreyscale(greyscaled);
+
+            var (intensityGradientsPic, direction) = intensityGradients(gaussianPic);
+
+            var nonMaxSupressedPic = nonMaxSupression(intensityGradientsPic, direction);
+
+            var doubleThresholdedPic = doubleThreshold(nonMaxSupressedPic);
+            var hysteresisedPic = hysteresis(doubleThresholdedPic);
+
+            return hysteresisedPic;

```

Comment:
```suggestion
            var (intensityGradientsPic, direction) = intensityGradients(BlurGreyscale(greyScale(pic)));
            return hysteresis(doubleThreshold(nonMaxSupression(intensityGradientsPic, direction)));

```
Vi har jo en guideline om function composition..?